### PR TITLE
SDO-338 feat: add github actions

### PR DESCRIPTION
**Pull request by AbrahamOEd**

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            git add open-fabricpayment.php
+
+            git checkout -b bump/v${new_version}
+            git commit -m "AUTOBUMP-${new_version} [skip ci]"
+            git push origin bump/v${new_version}
+            echo "Create new PR to bump version"
+            gh pr create --head bump/v${new_version} --base $(SOURCE_BRANCH) --fill
+
+            [ $? -eq 0 ] && export GH_TOKEN=${{ secrets.OF_GITHUB_TOKEN }}
+            [ $? -eq 0 ] && make .ci-helper-gh-auto-merge-pr-of-branch
```
Comment (0xGosu) [inline .github/workflows/main.yaml:69]: this make cmd helper will not be available here.

**Review by 0xGosu — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (AbrahamOEd) [reply .github/workflows/main.yaml:69]: my bad, just forgot to delete that line as the line underneath does what the make file does

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: api-openfabric
+          password: ${{ secrets.OF_GITHUB_TOKEN }}
+
+      - id: bump
+        name: Build and run Dev Container task
+        uses: devcontainers/ci@v0.3
```
Comment (0xGosu) [inline .github/workflows/main.yaml:45]: there is no devcontainer here, you need to use pure bash run job

**Review by 0xGosu — CHANGES_REQUESTED** [review]

**Review by 0xGosu — CHANGES_REQUESTED** [review]
pls check

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Checkout (GitHub)
+        uses: actions/checkout@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: api-openfabric
+          password: ${{ secrets.OF_GITHUB_TOKEN }}
```
Comment (0xGosu) [inline .github/workflows/main.yaml:41]: remove this

**Review by 0xGosu — CHANGES_REQUESTED** [review]

**Review by 0xGosu — CHANGES_REQUESTED** [review]
pls changes

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          sed -i "s/Version: $current_version/Version: $new_version/" openfabric-payment.php
+          echo "Bumped version from $current_version to $new_version"
+
+          git add open-fabricpayment.php
+
+          git checkout -b bump/v${new_version}
+          git commit -m "AUTOBUMP-${new_version} [skip ci]"
+          git push origin bump/v${new_version}
+          echo "Create new PR to bump version"
+          gh pr create --head bump/v${new_version} --base $(SOURCE_BRANCH) --fill
```
[the remark this answers is not in the document]
Comment (0xGosu) [reply .github/workflows/main.yaml:53]: pls fix this @AbrahamOEd

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: api-openfabric
+          password: ${{ secrets.OF_GITHUB_TOKEN }}
+
+      - id: bump
+        name: Build and run Dev Container task
```
[the remark this answers is not in the document]
Comment (0xGosu) [reply .github/workflows/main.yaml:37]: correct the name pls

**Review by 0xGosu — APPROVED** [review]

**Pull request merged**